sdk/storage/azure-storage-queue/tests/asyncqueuetestcase.py

```diff
@@ -0,0 +1,29 @@
+﻿# coding: utf-8
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import asyncio
+import functools
+from queuetestcase import QueueTestCase
+
+LOGGING_FORMAT = '%(asctime)s %(name)-20s %(levelname)-5s %(message)s'
+
+class AsyncQueueTestCase(QueueTestCase):
+    @staticmethod
+    def await_prepared_test(test_fn):
+        """Synchronous wrapper for async test methods. Used to avoid making changes
+        upstream to AbstractPreparer (which doesn't await the functions it wraps)
+        """
+
+        @functools.wraps(test_fn)
+        def run(test_class_instance, *args, **kwargs):
+            resource_group = kwargs.get("resource_group")
+            location = kwargs.get("location")
+            storage_account = kwargs.get("storage_account")
+            storage_account_key = kwargs.get("storage_account_key")
```

Comment:
You probably don't need to extract all these from kwargs, would it work just be doing:
```
loop.run_until_complete(test_funct(test_class_instance, **kwargs))
```